test/e2e/cronfederatedhpa_test.go

```diff
@@ -0,0 +1,154 @@
+			fhpa = helper.NewFederatedHPA(testNamespace, fhpaName, deploymentName)
+			cronFHPA = helper.NewCronFederatedHPAWithScalingFHPA(testNamespace, cronFHPAName, fhpaName, rule)
+
+			framework.CreateFederatedHPA(karmadaClient, fhpa)
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RemoveFederatedHPA(karmadaClient, testNamespace, fhpaName)
+			framework.RemoveCronFederatedHPA(karmadaClient, testNamespace, cronFHPAName)
+		})
+
+		ginkgo.It("Scale FederatedHPA testing", func() {
+			// Wait FederatedHPA to scale deployment's replicas to 1
+			framework.WaitDeploymentReplicasFitWith(framework.ClusterNames(), testNamespace, deploymentName, 1)
+
+			// Create CronFederatedHPA to scale FederatedHPA
+			framework.CreateCronFederatedHPA(karmadaClient, cronFHPA)
+
+			// Wait CronFederatedHPA to scale FederatedHPA's minReplicas to 2, which will trigger scaling deployment's replicas to 2
+			framework.WaitDeploymentReplicasFitWith(framework.ClusterNames(), testNamespace, deploymentName, 2)
+		})
+	})
+
+	// case 2. Scale deployment.
+	ginkgo.Context("Scale Deployment", func() {
+		ginkgo.BeforeEach(func() {
+			rule := helper.NewCronFederatedHPARule("scale-up", "*/1 * * * *", false, pointer.Int32(4), nil, nil)
+			cronFHPA = helper.NewCronFederatedHPAWithScalingDeployment(testNamespace, cronFHPAName, deploymentName, rule)
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RemoveCronFederatedHPA(karmadaClient, testNamespace, cronFHPAName)
+		})
+
+		ginkgo.It("Scale Deployment testing", func() {
+			// The init replicas of deployment should be 3
+			framework.WaitDeploymentReplicasFitWith(framework.ClusterNames(), testNamespace, deploymentName, 3)
+
+			// Create CronFederatedHPA to scale Deployment
+			framework.CreateCronFederatedHPA(karmadaClient, cronFHPA)
+
+			// Wait CronFederatedHPA to scale Deployment's replicas to 4
+			framework.WaitDeploymentReplicasFitWith(framework.ClusterNames(), testNamespace, deploymentName, 4)
+		})
+	})
+
+	// case 3. Test suspend rule with CronFederatedHPA
+	ginkgo.Context("Test suspend rule with CronFederatedHPA", func() {
+		ginkgo.BeforeEach(func() {
+			rule := helper.NewCronFederatedHPARule("scale-up", "*/1 * * * *", true, pointer.Int32(30), nil, nil)
+			cronFHPA = helper.NewCronFederatedHPAWithScalingDeployment(testNamespace, cronFHPAName, deploymentName, rule)
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RemoveCronFederatedHPA(karmadaClient, testNamespace, cronFHPAName)
+		})
+
+		ginkgo.It("Test suspend rule with CronFederatedHPA", func() {
+			// The init replicas of deployment should be 3
+			framework.WaitDeploymentReplicasFitWith(framework.ClusterNames(), testNamespace, deploymentName, 3)
+
+			// Create CronFederatedHPA to scale Deployment
+			framework.CreateCronFederatedHPA(karmadaClient, cronFHPA)
+
+			// Wait 3 minutes and check whether the replicas changed
+			time.Sleep(time.Minute*1 + time.Second*30)
```

Comment:
```suggestion
			time.Sleep(pollTimeout / 2)
```

Reply:
Why? The CronFederatedHPA is supposed to execute every minute. I will wait for 1:30 and check if the replicas have changed.

Reply:
I understand, but `(time.Minute*1 + time.Second*30)` is a magic number, so I suggest using pollTimeout instead.

Reply:
By the way,
wait `3 minutes`?  Waiting time seems too long which will seriously slow down the test progress.

Reply:
We should wait for this CronFederatredHPA rule, and check whether it's suspended, I didn't find other way to solve this.

Reply:
`pollTimeout = 420 * time.Second` is too long, I add some comments to make it clearly:
```
// */1 * * * * means the rule will be triggered every 1 minute
// So wait for 1m30s and check whether the replicas changed and whether the suspend field works
```

Reply:
@chaunceyjiang What do you think?
I didn't look into it.

Reply:
I also didn't find other way to solve this.

Reply:
Wait until can be a slightly more elegant way to modify it. Although fundamentally still need to wait for a little over a minute.